src/GraphQl/Resolver/Factory/CollectionResolverFactory.php

```diff
@@ -59,7 +59,7 @@ public function __construct(ReadStageInterface $readStage, SecurityStageInterfac
     public function __invoke(?string $resourceClass = null, ?string $rootClass = null, ?string $operationName = null): callable
     {
         return function (?array $source, array $args, $context, ResolveInfo $info) use ($resourceClass, $rootClass, $operationName) {
-            if (null === $resourceClass || null === $rootClass) {
+            if (null === $resourceClass || null === $rootClass || (null !== $source && !\array_key_exists($info->fieldName, $source))) {
```

Comment:
I think a comment would be helpful to understand what is going on here. Like:
If authorization has failed for a relation field, the field is not present in the source: null can be returned directly.